src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -224,10 +230,20 @@ final protected function isElementNameValid(string $name): bool
      */
     private function parseXml(\DOMNode $node, array $context = []): array|string
     {
+        $nodeName = $node->nodeName;
+
         $data = $this->parseXmlAttributes($node, $context);
 
         $value = $this->parseXmlValue($node, $context);
 
+        if (\is_array($value)
+            && ($childNodeName = $node->firstChild?->nodeName)
+            && 1 === \count($value)
+            && \in_array($nodeName, $context[self::FORCE_COLLECTION] ?? $this->defaultContext[self::FORCE_COLLECTION], true)
+        ) {
+            return [$childNodeName => [$value[$childNodeName]]];
+        }
```

Comment:
This condition can go after the `!\is_array($value)` if.
Then, you'll be able to drop the `is_array` check.

Reply:
Actually it can't according to the test suite 👀 
Looks like it actually passes the block below:
```php
if (!\count($data)) {
    return $value;
}
```